[6.0.0-rc.9] Universal schematics generating tsconfig in the wrong location for generated applications

### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Area
```
- [ ] devkit
- [x] schematics
```

### Versions
```sh
uname -a
Linux Jason-PC-U 4.13.0-39-generic #44-Ubuntu SMP Thu Apr 5 14:25:01 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
lsb_release -r
Release:        17.10
node -v
v8.11.1
npm -v
5.8.0
```

### Repro steps
<!--
Simple steps to reproduce this bug.
Please include: commands run, packages added, related code changes.
A link to a sample repo would help too.
-->
```sh
ng new proj
cd proj
ng g application newapp
ng g universal --client-project newapp
ng run newapp:server
```

### The log given by the failure
<!-- Normally this include a stack trace and some more information. -->
error TS5058: The specified path does not exist: '/home/jason/projects/work/temp/cli6proj/projects/newapp/src/tsconfig.app.json'.

The `tsconfig` file is put in the wrong directory for generated applications. It works fine for the project in the root since the tsconfig file is being put in the src directory. But for generated applications, the tsconfig files are in the actual root.

### Desired functionality
The tsconfig file generated by `ng g universal --client-app newapp` should be generated in the project root.
The tsconfig file generated for the application generated with `ng new` should still be generated in the `src` directory.


### Mention any other details that might be useful
<!-- Please include a link to the repo if this is related to an OSS project. -->

--- a/packages/schematics/angular/universal/files/root/__tsconfigFileName__.json
+++ b/packages/schematics/angular/universal/files/root/__tsconfigFileName__.json
@@ -6,6 +6,6 @@
     "module": "commonjs"
   },
   "angularCompilerOptions": {
-    "entryModule": "<%= appDir %>/<%= stripTsExtension(rootModuleFileName) %>#<%= rootModuleClassName %>"
+    "entryModule": "<%= rootInSrc ? '' : 'src/' %><%= appDir %>/<%= stripTsExtension(rootModuleFileName) %>#<%= rootModuleClassName %>"
   }
 }

--- a/packages/schematics/angular/universal/index.ts
+++ b/packages/schematics/angular/universal/index.ts
@@ -7,8 +7,10 @@
  */
 import {
   JsonObject,
+  Path,
   basename,
   experimental,
+  join,
   normalize,
   parseJson,
   strings,
@@ -67,7 +69,7 @@ function getClientArchitect(
   return clientArchitect;
 }
 
-function updateConfigFile(options: UniversalOptions): Rule {
+function updateConfigFile(options: UniversalOptions, tsConfigDirectory: Path): Rule {
   return (host: Tree) => {
     const workspace = getWorkspace(host);
     if (!workspace.projects[options.clientProject]) {
@@ -82,7 +84,7 @@ function updateConfigFile(options: UniversalOptions): Rule {
     const builderOptions: JsonObject = {
       outputPath: `dist/${options.clientProject}-server`,
       main: `${clientProject.root}src/main.server.ts`,
-      tsConfig: `${clientProject.root}src/tsconfig.server.json`,
+      tsConfig: join(tsConfigDirectory, `${options.tsconfigFileName}.json`),
     };
     const serverTarget: JsonObject = {
       builder: '@angular-devkit/build-angular:server',
@@ -214,26 +216,39 @@ export default function (options: UniversalOptions): Rule {
     const clientArchitect = getClientArchitect(host, options);
     const outDir = getTsConfigOutDir(host, clientArchitect);
     const tsConfigExtends = basename(clientArchitect.build.options.tsConfig);
+    const rootInSrc = clientProject.root === '';
+    const tsConfigDirectory = join(normalize(clientProject.root), rootInSrc ? 'src' : '');
 
     if (!options.skipInstall) {
       context.addTask(new NodePackageInstallTask());
     }
 
-    const templateSource = apply(url('./files'), [
+    const templateSource = apply(url('./files/src'), [
+      template({
+        ...strings,
+        ...options as object,
+        stripTsExtension: (s: string) => { return s.replace(/\.ts$/, ''); },
+      }),
+      move(join(normalize(clientProject.root), 'src')),
+    ]);
+
+    const rootSource = apply(url('./files/root'), [
       template({
         ...strings,
         ...options as object,
         stripTsExtension: (s: string) => { return s.replace(/\.ts$/, ''); },
         outDir,
         tsConfigExtends,
+        rootInSrc,
       }),
-      move(clientProject.root),
+      move(tsConfigDirectory),
     ]);
 
     return chain([
       mergeWith(templateSource),
+      mergeWith(rootSource),
       addDependencies(),
-      updateConfigFile(options),
+      updateConfigFile(options, tsConfigDirectory),
       wrapBootstrapCall(options),
       addServerTransition(options),
     ])(host, context);

--- a/packages/schematics/angular/universal/index_spec.ts
+++ b/packages/schematics/angular/universal/index_spec.ts
@@ -19,6 +19,9 @@ describe('Universal Schematic', () => {
   const defaultOptions: UniversalOptions = {
     clientProject: 'bar',
   };
+  const workspaceUniversalOptions: UniversalOptions = {
+    clientProject: 'workspace',
+  };
 
   const workspaceOptions: WorkspaceOptions = {
     name: 'workspace',
@@ -36,10 +39,22 @@ describe('Universal Schematic', () => {
     skipPackageJson: false,
   };
 
+  const initialWorkspaceAppOptions: ApplicationOptions = {
+    name: 'workspace',
+    projectRoot: '',
+    inlineStyle: false,
+    inlineTemplate: false,
+    routing: false,
+    style: 'css',
+    skipTests: false,
+    skipPackageJson: false,
+  };
+
   let appTree: UnitTestTree;
 
   beforeEach(() => {
     appTree = schematicRunner.runSchematic('workspace', workspaceOptions);
+    appTree = schematicRunner.runSchematic('application', initialWorkspaceAppOptions, appTree);
     appTree = schematicRunner.runSchematic('application', appOptions, appTree);
   });
 
@@ -57,9 +72,30 @@ describe('Universal Schematic', () => {
     expect(contents).toMatch(/export { AppServerModule } from '\.\/app\/app\.server\.module'/);
   });
 
-  it('should create a tsconfig file', () => {
+  it('should create a tsconfig file for the workspace project', () => {
+    const tree = schematicRunner.runSchematic('universal', workspaceUniversalOptions, appTree);
+    const filePath = '/src/tsconfig.server.json';
+    expect(tree.exists(filePath)).toEqual(true);
+    const contents = tree.readContent(filePath);
+    expect(JSON.parse(contents)).toEqual({
+      extends: './tsconfig.app.json',
+      compilerOptions: {
+        outDir: '../out-tsc/app-server',
+        baseUrl: '.',
+        module: 'commonjs',
+      },
+      angularCompilerOptions: {
+        entryModule: 'app/app.server.module#AppServerModule',
+      },
+    });
+    const angularConfig = JSON.parse(tree.readContent('angular.json'));
+    expect(angularConfig.projects.workspace.architect.server.options.tsConfig)
+      .toEqual('src/tsconfig.server.json');
+  });
+
+  it('should create a tsconfig file for a generated application', () => {
     const tree = schematicRunner.runSchematic('universal', defaultOptions, appTree);
-    const filePath = '/projects/bar/src/tsconfig.server.json';
+    const filePath = '/projects/bar/tsconfig.server.json';
     expect(tree.exists(filePath)).toEqual(true);
     const contents = tree.readContent(filePath);
     expect(JSON.parse(contents)).toEqual({
@@ -70,9 +106,12 @@ describe('Universal Schematic', () => {
         module: 'commonjs',
       },
       angularCompilerOptions: {
-        entryModule: 'app/app.server.module#AppServerModule',
+        entryModule: 'src/app/app.server.module#AppServerModule',
       },
     });
+    const angularConfig = JSON.parse(tree.readContent('angular.json'));
+    expect(angularConfig.projects.bar.architect.server.options.tsConfig)
+      .toEqual('projects/bar/tsconfig.server.json');
   });
 
   it('should add dependency: @angular/platform-server', () => {
@@ -93,7 +132,7 @@ describe('Universal Schematic', () => {
     const opts = arch.server.options;
     expect(opts.outputPath).toEqual('dist/bar-server');
     expect(opts.main).toEqual('projects/bar/src/main.server.ts');
-    expect(opts.tsConfig).toEqual('projects/bar/src/tsconfig.server.json');
+    expect(opts.tsConfig).toEqual('projects/bar/tsconfig.server.json');
   });
 
   it('should add a server transition to BrowerModule import', () => {

